### PR TITLE
Make Tooltips Accessible

### PR DESCRIPTION
**Pull request by iamsolankiamit**
Change-type: Minor
Signed-off-by: Amit Solanki <amit@balena.io>

---
##### Contributor checklist

- [ ] I have rebuilt the README with `npm run build:docs`
- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [x] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: src/components/Badge/index.tsx [hunk 1]
```diff
@@ -46,7 +46,12 @@ const BaseBadge = styled(Txt.span)`
 	line-height: 1;
 `;
 
-const Badge = ({ children, className, shade }: ThemedBadgeProps) => {
+const Badge = ({
```
Comment (LucianBuzzo) [inline src/components/Badge/index.tsx]: This refactor should be in a separate commit, or even better, a separate PR.

File: src/tooltips.tsx [hunk 2]
```diff
@@ -13,25 +13,6 @@ const TARGET_OFFSET = 3;
 // The time in ms to show a tooltip after being triggered by a click event
 const CLICK_TIMEOUT = 1000;
 
-const createOnRemoveObserver = (
```
Comment (LucianBuzzo) [inline src/tooltips.tsx]: This refactor should be in a separate commit, or even better, a separate PR.

Comment (iamsolankiamit) [reply src/tooltips.tsx]: @LucianBuzzo This refactor is required, in the previous implementation we only rendered one tooltip component and updated it's content based on the hovered/clicked component. for accessibility changes to work, we need to render all the tooltips in separate component and connect them to their references using id and aria-describedby.

File: src/components/Badge/index.tsx (hunk 1 above)
Comment (iamsolankiamit) [reply src/components/Badge/index.tsx]: Sure, I can move it into a separate commit.

File: src/tooltips.tsx (hunk 2 above)
Comment (LucianBuzzo) [reply src/tooltips.tsx]: @iamsolankiamit My comment stands, if it is a pre-requisite for making the tooltips accessible, then it should be made in a sperate commit and labeled as such

**Comment by iamsolankiamit** [comment]
closing this in favour of #1068
